rse-py-package.Rmd

```diff
@@ -0,0 +1,1460 @@
+only those things (where versions come from `pip freeze`):
+
+<!-- src="rse-py-package/10/requirements_docs.txt" -->
+```text
+Sphinx>=1.7.4
+m2r==0.2.1
+```
+
+Then anyone wanting to build the documentation (including you, on another computer)
+need only run `pip install -r requirement_docs.txt`
+
+This breakdown of requirements by use case is often seen with packages
+that have a `requirements_dev.txt` for additional packaged not needed
+for users of the package, but for its development and testing.  For this
+package, we could usefully pull out `pytest` from our `requirements.txt`
+into a `requirements_dev.txt`
+
+## How do I put the documentation for my package? online {#rse-py-package-rtd}
+
+There are a number of ways of hosting the web pages that document
+your project.  A very common option, particularly for python
+projects, is [Read The Docs](https://readthedocs.org), a community
+supported site which hosts software documentation free of charge.
+
+Read The Docs integrates nicely with GitHub, which will allow
+you to have the documentation re-built automatically upon update
+of your repository.  Thus we recommend 
+[registering for Read The Docs](https://readthedocs.org/accounts/signup/) with your GitHub
+account.
+
+Once you have registered for Read The Docs, confirmed your email, and signed in,
+you will be able to add new projects by [importing a project](https://readthedocs.org/dashboard/import/?)
+from your github repository; Read The Docs will then build the documentation
+(_e.g._, run make) and host the resulting files.  For this to work, all of
+the source files (in our case, `docs/source/modules.rst`, `docs/source/zipfpy.rst`,
+`docs/Makefile`, `docs.conf.py`) need to be checked into your repository.
+
+Once this is done, you can import your repository; in this case since there
+are additional requirements (the `m2r` package), you will have to make sure
+Read The Docs installs this when building the documentation; when importing,
+under "Advanced Settings", set the "Requirements file" to be `requirements_docs.txt`.
+Then you should be able to build; after a few minutes, your documents should 
+be visible!  If you named your project, _e.g._, `pterry_zipfpy`, your 
+documents will be available at 
+[`https://pterry_zipfpy.readthedocs.io/en/latest/`](https://pterry_zipfpy.readthedocs.io/en/latest/).
+
+FIXME: Should we just use the configuration file approach here rather than going
+into advanced settings?
+
+As with continuous integration testing, there are badges for your README linking
+both to the documentation and reporting on the status of the documentation build;
+you would be able to add that badge and link by adding the following line to your
+`README.md`:
+
+```text
+[![Documentation Status](https://readthedocs.org/projects/pterry_zipfpy/badge/?version=latest)](https://pterry_zipfpy.readthedocs.io/en/latest/?badge=latest)
+```
+
+FIXME: talk about the webhooks, bringing in the discussion from the CI section.
```

Comment:
file issues to make sure it's covered there?

Reply:
It is now, and have added the appropriate text and reference.